profiler/internal/fastdelta/fd_test.go

```diff
@@ -0,0 +1,440 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package fastdelta
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/google/pprof/profile"
+	"github.com/richardartoul/molecule"
+	"github.com/richardartoul/molecule/src/protowire"
+	"github.com/stretchr/testify/require"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/profiler/internal/pprofutils"
+)
+
+const heapFile = "heap.pprof"
+const bigHeapFile = "big-heap.pprof"
+
+func BenchmarkFastDelta(b *testing.B) {
+	for _, f := range []string{heapFile, bigHeapFile} {
+		testFile := "testdata/" + f
+		b.Run(testFile, func(b *testing.B) {
+			b.ReportAllocs()
+			before, err := os.ReadFile(testFile)
+			if err != nil {
+				b.Fatal(err)
+			}
+			after, err := os.ReadFile(testFile)
+			if err != nil {
+				b.Fatal(err)
+			}
+			b.ResetTimer()
+
+			buf := new(bytes.Buffer)
+			for i := 0; i < b.N; i++ {
+				buf.Reset()
+				dc := NewDeltaComputer(
+					vt("alloc_objects", "count"),
+					vt("alloc_space", "bytes"),
+				)
+				err := dc.Delta(before, io.Discard)
+				if err != nil {
+					b.Fatal(err)
+				}
+				err = dc.Delta(after, buf)
+				if err != nil {
+					b.Fatal(err)
+				}
```

Comment:
I agree. We could re-work this and even incorporate `b.N` into the test input.  I don't think that would be too difficult.

> It doesn't measure any delta computation.

Do you mean the [one actual delta in the bench](https://github.com/DataDog/dd-trace-go/blob/1da2b74838c81b4038a5fe64cb3d6a92a778361e/profiler/internal/fastdelta/fd_test.go#L52) results in no non-zero samples for the [value types](https://github.com/DataDog/dd-trace-go/blob/1da2b74838c81b4038a5fe64cb3d6a92a778361e/profiler/internal/fastdelta/fd_test.go#L45-L46) configured?  The table you have is only showing the set of non-`alloc_*` samples.

Reply:
As there is only one `Delta` computation, the current benchmark is under-representing the gains for processes that last longer than 2 profiling cycles.

Reply:
> As there is only one `Delta` computation, the current benchmark is under-representing the gains for processes that last longer than 2 profiling cycles.

Yes, that's another problem with this benchmark! I think it'd be better if it create a single DeltaComputer and then applied pprofs against it in the loop rather than creating a new computer in each loop.

Reply:
Agreed.  This bench could use some more work.  I'll have some updates later today.